src/interfaces/permissions/types.ts

```diff
@@ -0,0 +1,30 @@
+export type Scope = {
+  method: string
+  schema?: string
+  objectId?: string
+};
+
+export type Conditions = {
+  // attestation indicates whether any inbound data should be signed.
+  // defaults to `optional`
```

Comment:
All the "defaults to <xyz>" are weird here because the code doesn't/can't follow the requirement here. Consider remove.

Reply:
This `Conditions` typed has to get canned anyway because `Scope` and `Conditions` are now specific to interface methods ☹️